lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1314,20 +1318,58 @@ def _regenerate_alert_rules(self):
         self._generate_alert_rules_files(self.container)
         logger.debug("Saved alerts rules to disk")
 
-    def _endpoints(self) -> List[dict]:
-        """Return a list of Loki Push Api endpoints."""
-        return [{"url": self._url(unit_number=i)} for i in range(self._charm.app.planned_units())]
+    def update_endpoint(self, url: str = None, relation: Relation = None) -> None:
+        """Triggers programmatically the update of endpoint in unit relation data.
+
+        This method should be used when the charm relying on this library needs
+        to update the relation data in response to something occurring outside
+        of the `logging` relation lifecycle, e.g., in case of a
+        host address change because the charmed operator becomes connected to an
+        Ingress after the `logging` relation is established.
+
+        Args:
+            url: An optional url value to update relation data.
+            relation: An optional instance of `class:ops.model.Relation` to update.
+        """
+        if not relation:
+            if not self._charm.model.get_relation(self._relation_name):
+                return
+
+            relation = self._charm.model.get_relation(self._relation_name)
```

Comment:
```suggestion
        relation = relation or self._charm.model.get_relation(self._relation_name)
        if not relation:
            return
```

Reply:
If we do this we are going to override the relation that can be passed to the function